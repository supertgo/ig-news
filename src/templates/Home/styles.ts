import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 112rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.medium};
    height: calc(100vh- 8rem);

    display: flex;
    align-items: center;
    justify-content: space-between;
  `}
`;

export const Image = styled.img``;

export const Section = styled.section`
  max-width: 60rem;

  button {
    margin-top: 4rem;
  }
`;

export const Text = styled.span`
  ${({ theme }) => css`
    font-size: 2.4rem;
    font-weight: ${theme.font.bold};
  `}
`;

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: 7.2rem;
    line-height: 7.2rem;
    font-weight: 900;
    margin-top: ${theme.spacings.large};
  `}
`;

export const Span = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.blue};
  `}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    font-size: 2.4rem;
    line-height: 3.6rem;

    span {
      font-weight: ${theme.font.bold};
    }
  `}
`;
