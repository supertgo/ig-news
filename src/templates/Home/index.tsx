import Header from 'components/Header';
import Head from 'next/head';

import * as S from './styles';

const Home = () => (
  <>
    <Head>
      <title>Inicio | ig.news</title>
    </Head>
    <Header />
    <S.Wrapper>
      <S.Section>
        <S.Text>üëè Hey, welcome</S.Text>
        <S.Title>
          News about the <S.Span>React</S.Span> world.
        </S.Title>
        <S.Content>
          Get access to all the publications <br />
          <S.Span>for $9.90 month</S.Span>
        </S.Content>
      </S.Section>
      <S.Image src="/images/avatar.svg" alt="Girl coding" />
    </S.Wrapper>
  </>
);

export default Home;
