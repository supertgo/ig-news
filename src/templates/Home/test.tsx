import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Home from '.';

jest.mock('components/Header', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Header"></div>;
    }
  };
});

describe('<Home />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Home />);

    expect(screen.getByText(/Hey, welcome/i)).toBeInTheDocument();
    expect(screen.getByText(/News about the/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Get access to all the publications/i)
    ).toBeInTheDocument();

    expect(
      screen.getByRole('img', { name: /Girl coding/i })
    ).toBeInTheDocument();

    expect(screen.getByTestId('Mock Header')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
