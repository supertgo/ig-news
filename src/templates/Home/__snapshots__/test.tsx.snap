// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the heading 1`] = `
<div
  data-testid="Mock Header"
/>
`;
